app/lib/accounts.js

```diff
@@ -16,6 +16,25 @@ AccountsTemplates.configureRoute("forgotPwd");
 AccountsTemplates.configureRoute("resetPwd");
 AccountsTemplates.configureRoute("signIn");
 
-Meteor.startup(function() {
+Meteor.startup(function () {
+  Accounts.validateNewUser(function (user) {
+    if (!user) return false
+    if (!user.emails || (!!user.emails && user.emails.length === 0)) return false
+    const emailValidationRegex = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    /* Iterates through all emails and checks if all emails are valid */
+    const areAllEmailsValid = user.emails.every(emailObject => {
```

Comment:
Lets try to use Simple Schema here, so the code is less verbose.

https://github.com/aldeed/simple-schema-js/blob/master/README.md#validate-an-array-of-objects-and-throw-an-error

https://github.com/aldeed/simple-schema-js/blob/master/README.md#regex